code/WorkInProgress/MechanicMadness.dm

```diff
@@ -977,6 +977,38 @@ var/list/mechanics_telepads = new/list()
 		icon_state = "[under_floor ? "u":""]comp_accel"
 		return
 
+/obj/item/mechanics/zapper
+	name = "Tesla Coil"
+	desc = ""
+	icon_state = "comp_zap"
+	cooldown_time = 1 SECONDS
+	var/zap_power = 2
+	var/zap_radius = 0
+
+	New()
+		..()
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_INPUT,"zap", "eleczap")
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_CONFIG,"Set Power","setPower")
+
+	proc/eleczap(var/datum/mechanicsMessage/input)
```

Comment:
this should use power from the power grid imo

Reply:
Ive though about this, but neither multitools, nor the vast majority of the mechanics components consume power. You are already giving away the portability of the mutitool in exchange for the automation, no need to give it another downside.

Reply:
I don't know that requiring engine power would necessarily be a downside. Giving it access to a 500TW engine Hotwire would be....interesting